ci/test-sriov-secondary-network-aws.sh

```diff
@@ -0,0 +1,552 @@
+verify_cluster() {
+    local control_plane_node_ip=$1
+    echo "Verifying Kubernetes cluster status on control-plane node $control_plane_node_ip..."
+
+    ssh -o StrictHostKeyChecking=no -i "$AWS_EC2_SSH_KEY_NAME" ubuntu@$control_plane_node_ip << EOF
+        kubectl get nodes
+EOF
+}
+
+function setup_cluster {
+    # Launch Master and Worker EC2 instances
+    echo "Launching EC2 instance ..."
+    CONTROLPLANE_INSTANCE_ID=$(launch_ec2_instance "ControlPlane")
+    echo "ControlPlane EC2 instance launched with Instance ID: $CONTROLPLANE_INSTANCE_ID"
+    WORKER_INSTANCE_ID=$(launch_ec2_instance "Worker")
+    echo "Worker EC2 instance launched with Instance ID: $WORKER_INSTANCE_ID"
+
+    attach_network_interface "$CONTROLPLANE_INSTANCE_ID" "control-plane"
+    attach_network_interface "$WORKER_INSTANCE_ID" "worker"
+
+    echo "======== CONTROLPLANE_INSTANCE_ID: $CONTROLPLANE_INSTANCE_ID, WORKER_INSTANCE_ID: $WORKER_INSTANCE_ID ========="
+
+    # Get the public IP addresses of the instances
+    CONTROLPLANE_IP=$(get_instance_ip "$CONTROLPLANE_INSTANCE_ID")
+    echo "Get PublicIpAddress of EC2 Instance $CONTROLPLANE_INSTANCE_ID: $CONTROLPLANE_IP"
+    WORKER_IP=$(get_instance_ip "$WORKER_INSTANCE_ID")
+    echo "Get PublicIpAddress of EC2 Instance $WORKER_INSTANCE_ID: $WORKER_IP"
+
+    echo "====== CONTROLPLANE_IP: $CONTROLPLANE_IP, WORKER_IP: $WORKER_IP ======"
+
+    # Wait for EC2 instances to be fully ready
+    echo "Waiting for EC2 instances to be ready..."
+    sleep 120  # You may adjust the sleep time based on your environment
+
+    # Install Kubernetes on both control-plane and worker nodes
+    install_kubernetes "$CONTROLPLANE_IP"
+    install_kubernetes "$WORKER_IP"
+
+    # Initialize Kubernetes control-plane Node, if it fails, it will not exit
+    initialize_control_plane_node "$CONTROLPLANE_IP" || {
+        echo "Failed to initialize control-plane Node. Re-running installation for both nodes."
+        # Re-run the install commands for both control-plane and worker nodes
+        install_kubernetes "$CONTROLPLANE_IP"
+        install_kubernetes "$WORKER_IP"
+        # After re-installing, attempt the initialization again
+        initialize_control_plane_node "$CONTROLPLANE_IP" || {
+            echo "Initialization ControlPlane failed again, please check logs."
+            exit 1
+        }
+    }
+
+    # Get the join command and join the worker node to the cluster
+    JOIN_COMMAND=$(get_join_command "$CONTROLPLANE_IP")
+    join_worker "$WORKER_IP" "$JOIN_COMMAND"
+
+    # Verify the Kubernetes cluster status
+    verify_cluster "$CONTROLPLANE_IP"
+
+    echo "Kubernetes cluster setup completed!"
+}
+
+build_image() {
+    chmod -R g-w build/images/ovs
+    chmod -R g-w build/images/base
+    ./hack/build-antrea-linux-all.sh --pull
+    TEMP_ANTREA_TAR="antrea-ubuntu.tar"
+    sudo docker save antrea/antrea-agent-ubuntu:latest antrea/antrea-controller-ubuntu:latest -o $TEMP_ANTREA_TAR
```

Comment:
you should not be using `sudo` here
`./hack/build-antrea-linux-all.sh` will already run docker commands without `sudo` anyway

more generally, do not use `sudo` for any command that runs _locally_

Reply:
Updated.